kong/clustering/services/sync/rpc.lua

```diff
@@ -176,7 +177,23 @@ local function is_rpc_ready()
 end
 
 
-local function do_sync()
+local do_sync
+
+
+local function error_handler(err, is_full_sync)
+  ngx_log(ngx_ERR, err)
+
+   -- retry without releasing the mutex as the DP has nothing to do except syncing
+  if is_full_sync then
+    yield()
+    return do_sync()
+  end
+
+  return nil, err
+end
+
+
+function do_sync()
```

Comment:
This is equivalent to `_G.do_sync = function() ... end` instead of `do_sync = function() .. end`.

Reply:
```
$ luajit
> local foo
> print(tostring(foo))
nil
> function foo() end
> print(tostring(foo))
function: 0x745fa021aa70
> print(tostring(_G.foo))
function: 0x745fa021aa70
> 
```

Reply:
It's my fault; it doesn't assign to `_G` outside the interactive interpreter.